### PR TITLE
combine share and data into custom-controls component

### LINKED ISSUE #145: Custom map control popups (share, data) should close each other
When opening share or data, the other should close. These controls should share a class. The "Share Controls" SASS module should be renamed generically for both cases.

### PR DESCRIPTION
**Pull request by andycochran**
This PR… 
- Adds a new `custom-controls` component which combines `share-controls`, `data-controls` and `print-controls` so that they are aware of each other.  This is so that they're aware of each other and each will close when the other is opened. Closes #145. 
- CSS and classes are also abstracted to work for all components, instead of repeating. Each control only needs `position-top` and `z-index`. 
- Tests are improved (there was previously no test for the data controls).

### CODE REVIEW

File: app/components/custom-controls.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import Component from '@ember/component';
+import { argument } from '@ember-decorators/argument';
+import { action, computed } from '@ember-decorators/object';
+import { run } from '@ember/runloop';
+
+export default class ShareControls extends Component {
```
Comment (allthesignals) [inline app/components/custom-controls.js]: I would rename to CustomControls

File: app/components/custom-controls.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  // Share Control properties
+  @argument shareURL;
+  @argument shareClosed = true;
+  @argument copySuccess = false;
+
+  // Data Control properties
+  @argument dataClosed = true;
+  @argument boundsGeoJSON;
+
+  datasets = [
```
Comment (allthesignals) [inline app/components/custom-controls.js]: Should this also be an `@argument` with this default?

Comment (andycochran) [reply app/components/custom-controls.js]: Class updated. 

What change are you proposing to the argument?

**Comment by allthesignals** [comment]
Thanks @andycochran !!!

**Pull request merged**